inc/azure_uamqp_c/session.h

```diff
@@ -43,24 +45,11 @@ DEFINE_ENUM(SESSION_SEND_TRANSFER_RESULT, SESSION_SEND_TRANSFER_RESULT_VALUES)
     typedef bool(*ON_LINK_ATTACHED)(void* context, LINK_ENDPOINT_HANDLE new_link_endpoint, const char* name, role role, AMQP_VALUE source, AMQP_VALUE target);
 
     MOCKABLE_FUNCTION(, SESSION_HANDLE, session_create, CONNECTION_HANDLE, connection, ON_LINK_ATTACHED, on_link_attached, void*, callback_context);
-    MOCKABLE_FUNCTION(, SESSION_HANDLE, session_create_from_endpoint, CONNECTION_HANDLE, connection, ENDPOINT_HANDLE, connection_endpoint, ON_LINK_ATTACHED, on_link_attached, void*, callback_context);
     MOCKABLE_FUNCTION(, int, session_set_incoming_window, SESSION_HANDLE, session, uint32_t, incoming_window);
     MOCKABLE_FUNCTION(, int, session_get_incoming_window, SESSION_HANDLE, session, uint32_t*, incoming_window);
     MOCKABLE_FUNCTION(, int, session_set_outgoing_window, SESSION_HANDLE, session, uint32_t, outgoing_window);
     MOCKABLE_FUNCTION(, int, session_get_outgoing_window, SESSION_HANDLE, session, uint32_t*, outgoing_window);
-    MOCKABLE_FUNCTION(, int, session_set_handle_max, SESSION_HANDLE, session, handle, handle_max);
-    MOCKABLE_FUNCTION(, int, session_get_handle_max, SESSION_HANDLE, session, handle*, handle_max);
```

Comment:
>MOCKABLE_FUNCTION [](start = 4, length = 17)

I don't quite like this split in internal and non internal for link/session/message.
I see no need for splitting things. Generally I do not like a layer of internals because it only complicates things. The link APIs are all available to its consumer and if a user decides to use the SDK at this link level, all power to them.

So please back it out. Pretty please.

Reply:
Same applies to session, message.

---
In reply to: [168280844](https://github.com/Azure/azure-uamqp-c/pull/212#discussion_r168280844) [](ancestors = 168280844)

Reply:
Since I'd like to see this backed out, I will wait for it.

---
In reply to: [168280940](https://github.com/Azure/azure-uamqp-c/pull/212#discussion_r168280940) [](ancestors = 168280940,168280844)

Reply:
Agreed per our discussion.  Next push will incorporate this back-out.

---
In reply to: [168282285](https://github.com/Azure/azure-uamqp-c/pull/212#discussion_r168282285) [](ancestors = 168282285,168280940,168280844)